src/plugins/defoldcollection/defoldcollectionplugin.cpp

```diff
@@ -0,0 +1,473 @@
+#include "tile.h"
+#include "tilelayer.h"
+#include "grouplayer.h"
+
+#include <QTextStream>
+#include <QFileInfo>
+#include <QDir>
+
+#include <cmath>
+
+namespace DefoldCollection {
+
+static const char cell_t[] =
+"  cell {\n\
+    x: {{x}}\n\
+    y: {{y}}\n\
+    tile: {{tile}}\n\
+    h_flip: {{h_flip}}\n\
+    v_flip: {{v_flip}}\n\
+  }\n";
+
+static const char layer_t[] =
+"layers {\n\
+  id: \"{{id}}\"\n\
+  z: {{z}}\n\
+  is_visible: {{is_visible}}\n\
+{{cells}}\
+}\n";
+
+static const char map_t[] =
+"tile_set: \"{{tile_set}}\"\n\
+{{layers}}\n\
+material: \"{{material}}\"\n\
+blend_mode: {{blend_mode}}\n";
+
+static const char collection_t[] =
+"name: \"default\"\n\
+scale_along_z: 0\n\
+embedded_instances {\n\
+  id: \"tilemaps\"\n\
+{{children}}\
+  data: {{components}}\n\
+  \"\"\n\
+  position {\n\
+    x: {{pos-x}}\n\
+    y: {{pos-y}}\n\
+    z: 0.0\n\
+  }\n\
+  rotation {\n\
+    x: 0.0\n\
+    y: 0.0\n\
+    z: 0.0\n\
+    w: 1.0\n\
+  }\n\
+  scale3 {\n\
+    x: 1.0\n\
+    y: 1.0\n\
+    z: 1.0\n\
+  }\n\
+}\n\
+{{embedded-instances}}\n\
+\n";
+
+static const char component_t[] =
+"  \"components {\\n\"\n\
+  \"  id: \\\"{{tilemap_name}}\\\"\\n\"\n\
+  \"  component: \\\"{{tilemap_rel_path}}\\\"\\n\"\n\
+  \"  position {\\n\"\n\
+  \"    x: 0.0\\n\"\n\
+  \"    y: 0.0\\n\"\n\
+  \"    z: 0.0\\n\"\n\
+  \"  }\\n\"\n\
+  \"  rotation {\\n\"\n\
+  \"    x: 0.0\\n\"\n\
+  \"    y: 0.0\\n\"\n\
+  \"    z: 0.0\\n\"\n\
+  \"    w: 1.0\\n\"\n\
+  \"  }\\n\"\n\
+  \"}\\n\"\n\
+";
+
+static const char child_t[] =
+"  children: \"{{child-name}}\"\n\
+";
+
+static const char emdedded_instance_t[] =
+"embedded_instances {\n\
+  id: \"{{instance-name}}\"\n\
+  data: {{components}}\n\
+  \"\"\n\
+  position {\n\
+    x: 0.0\n\
+    y: 0.0\n\
+    z: 0.0\n\
+  }\n\
+  rotation {\n\
+    x: 0.0\n\
+    y: 0.0\n\
+    z: 0.0\n\
+    w: 1.0\n\
+  }\n\
+  scale3 {\n\
+    x: 1.0\n\
+    y: 1.0\n\
+    z: 1.0\n\
+  }\n\
+}\n\
+";
+
+static QString replaceTags(QString context, const QVariantHash &map)
+{
+    QHashIterator<QString,QVariant> it{map};
+    while (it.hasNext()) {
+        it.next();
+        context.replace(QLatin1String("{{") + it.key() + QLatin1String("}}"),
+                        it.value().toString());
+    }
+    return context;
+}
+
+DefoldCollectionPlugin::DefoldCollectionPlugin()
+{
+}
+
+QString DefoldCollectionPlugin::nameFilter() const
+{
+    return tr("Defold collection (*.collection)");
+}
+
+QString DefoldCollectionPlugin::shortName() const
+{
+    return QLatin1String("defoldcollection");
+}
+
+QString DefoldCollectionPlugin::errorString() const
+{
+    return mError;
+}
+
+/*
+ * Returns a new filepath relative to the root of the Defold project if we're in one.
+ * Determines the root of the project by looking for a file called "game.project".
+ * If no such file is found by going up the heirarchy, return filename from the filepath.
+*/
+QString DefoldCollectionPlugin::TilesetRelativePath(QString filePath)
```

Comment:
Coding style: please rename to `tilesetRelativePath`.

Also please pass `QString` as `const QString &` when possible.

Finally I think this function could be a static local function instead of a member.